tools/backups2datalad/datasetter.py

```diff
@@ -137,7 +137,19 @@ async def update_dandiset(
                 create_time=dandiset.version.created,
             )
         dmanager = self.manager.with_sublogger(f"Dandiset {dandiset.identifier}")
-        changed, zarr_stats = await self.sync_dataset(dandiset, ds, dmanager)
+        state = ds.get_assets_state()
+        if state is None or state.timestamp < dandiset.version.modified:
+            changed, zarr_stats = await self.sync_dataset(dandiset, ds, dmanager)
```

Comment:
could you please add  for paranoid me:
```suggestion
            changed, zarr_stats = await self.sync_dataset(dandiset, ds, dmanager)
        elif state.timestamp > dandiset.version.modified:
            raise ValueError(f"BackToThePastError: Remote dandiset {dandiset.identifier} has modification datetime {dandiset.version.modified} BEFORE currently recorded {state.timestamp}")
```
or alike

Reply:
Done: b5de111